src/mito2/src/sst/parquet/file_range.rs

```diff
@@ -222,40 +262,50 @@ impl FileRangeContext {
 
     /// TRY THE BEST to perform pushed down predicate precisely on the input batch.
     /// Return the filtered batch. If the entire batch is filtered out, return None.
-    pub(crate) fn precise_filter(&self, input: Batch) -> Result<Option<Batch>> {
-        self.base.precise_filter(input)
+    pub(crate) fn precise_filter(&self, input: Batch, skip_fields: bool) -> Result<Option<Batch>> {
+        self.base.precise_filter(input, skip_fields)
     }
 
     /// Filters the input RecordBatch by the pushed down predicate and returns RecordBatch.
-    pub(crate) fn precise_filter_flat(&self, input: RecordBatch) -> Result<Option<RecordBatch>> {
-        self.base.precise_filter_flat(input)
+    pub(crate) fn precise_filter_flat(
+        &self,
+        input: RecordBatch,
+        skip_fields: bool,
+    ) -> Result<Option<RecordBatch>> {
+        self.base.precise_filter_flat(input, skip_fields)
+    }
+
+    /// Determines whether to skip field filters based on PreFilterMode and row group delete status.
+    pub(crate) fn should_skip_fields(&self, row_group_idx: usize) -> bool {
+        match self.base.pre_filter_mode {
+            PreFilterMode::All => false,
+            PreFilterMode::SkipFields => true,
+            PreFilterMode::SkipFieldsOnDelete => {
+                // Check if this specific row group contains delete op
+                self.contains_delete(row_group_idx).unwrap_or(true)
+            }
+        }
     }
 
     //// Decodes parquet metadata and finds if row group contains delete op.
     pub(crate) fn contains_delete(&self, row_group_index: usize) -> Result<bool> {
         let metadata = self.reader_builder.parquet_metadata();
-        let row_group_metadata = &metadata.row_groups()[row_group_index];
-
-        // safety: The last column of SST must be op_type
-        let column_metadata = &row_group_metadata.columns().last().unwrap();
-        let stats = column_metadata.statistics().context(StatsNotPresentSnafu {
-            file_path: self.reader_builder.file_path(),
-        })?;
-        stats
-            .min_bytes_opt()
-            .context(StatsNotPresentSnafu {
-                file_path: self.reader_builder.file_path(),
-            })?
-            .try_into()
-            .map(i32::from_le_bytes)
-            .map(|min_op_type| min_op_type == OpType::Delete as i32)
-            .ok()
-            .context(DecodeStatsSnafu {
-                file_path: self.reader_builder.file_path(),
-            })
+        row_group_contains_delete(metadata, row_group_index, self.reader_builder.file_path())
     }
 }
 
+/// Mode to pre-filter columns in a range.
+#[derive(Debug, Clone, Copy)]
+pub enum PreFilterMode {
```

Comment:
Looks likje `FilterApplyMode` would be more understandable?

Reply:
We may do a post filter after the merge reader later so I named it as PreFilterMode.